src/check.h.in

```diff
@@ -669,6 +671,396 @@ CK_DLL_EXP void CK_EXPORT _ck_assert_failed(const char *file, int line,
+ * compared to ck_assert(). */
+/* OP can have values: <; >=. */
+#define _ck_assert_floating_absdiff_op_tol10(X, Y, OP, P) \
+do { \
+  long double _ck_x = (X); \
+  long double _ck_y = (Y); \
+  int _ck_p = (P); \
+  _ck_assert_floating_tol10(P); \
+  ck_assert_msg(fabsl(_ck_y - _ck_x) OP powl(10, _ck_p), \
+    "Assertion '%s' failed: %s == %.*Lg, %s == %.*Lg, %s == %d", \
+    "fabsl("#Y" - "#X") "#OP" powl(10, "#P")", \
+    #X, (int)CK_FLOATING_DIG, _ck_x, \
+    #Y, (int)CK_FLOATING_DIG, _ck_y, \
+    #P, _ck_p); \
+} while (0)
+
+/**
+ * Check two floating point numbers to determine if X≈Y with specified tolerance
+ * level
+ *
+ * If not X ≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_eq_tol10(X, Y, P)  _ck_assert_floating_absdiff_op_tol10(X, Y, <, P)
+
+/**
+ * Check two floating point numbers to determine if not X≈Y with specified
+ * tolerance level
+ *
+ * If X ≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_ne_tol10(X, Y, P) _ck_assert_floating_absdiff_op_tol10(X, Y, >=, P)
+
+/**
+ * Check two floating point numbers to determine if X>≈Y with specified
+ * tolerance level
+ *
+ * If not X >≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_ge_tol10(X, Y, P) _ck_assert_floating_op_tol10(X, >, Y, P, -1)
+
+/**
+ * Check two floating point numbers to determine if X<≈Y with specified
+ * tolerance level
+ *
+ * If not X <≈ Y with error < powl(10, P), the test fails.
+ *
+ * @param X floating point number
+ * @param Y floating point number to compare against X
+ * @param P power of 10 (integer) as tolerance, if negative number is used
+ * it would mean number of decimal places (precision)
+ *
+ * @note If the check fails, the remaining of the test is aborted
+ *
+ * @since 0.11.0
+ */
+#define ck_assert_floating_le_tol10(X, Y, P) _ck_assert_floating_op_tol10(X, <, Y, P, 1)
+
+/**
+ * Check floating point number to be finite (none of those: +infinity,
```

Comment:
Check that a floating point number is finite; i.e. is not +infinity, -infinity, or NaN.